sky/clouds/service_catalog/common.py

```diff
@@ -270,6 +270,25 @@ def get_vcpus_from_instance_type_impl(
     return float(vcpus)
 
 
+def _filter_with_cpu(df: pd.DataFrame, cpu: Optional[str]) -> pd.DataFrame:
+    if cpu is None:
+        return df
+    if cpu.endswith('+'):
+        return df[df['vCPUs'] >= float(cpu[:-1])]
+    else:
+        return df[df['vCPUs'] == float(cpu)]
```

Comment:
nit: To be safer, maybe catch ValueError again?

Reply:
I think we shouldn't raise the error when the empty dataframe is returned, because there might exist other clouds that matches the requested number of vCPUs. If all clouds do not have a matching VM, the optimizer will then print the error msg.

Reply:
Oh, I meant ValueError from parsing `float(cpu...)`. A safeguard to remove the assumption that arg `cpu` is validated upstream.

Reply:
I feel it's redundant. In which case do you think we'll need it?

Reply:
E.g., in the future somehow the service_catalog function gets called elsewhere.

Reply:
Alright. Added.